### PR TITLE
Update tests for PHPUnit 6

### PR DESCRIPTION
**Pull request by padraic**
#26 will need to be rebased if this is merged as it contains one extra testcase file.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
         "padraic/humbug_get_contents": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.0"
+        "phpunit/phpunit": "~6.0"
```
Comment (xabbuh) [inline composer.json]: This means that you cannot test the library with any PHP version before 7.0. Doesn't it make sense to drop support for these PHP version then?

Comment (padraic) [inline composer.json]: You're right, I need to allow for a lesser version to resolve deps for composer on Travis for PHP 5.6.

Comment (xabbuh) [inline composer.json]: `^5.7 || ^6.0` should do the trick if you want to drop support for PHP < 5.6

**Comment by padraic** [comment]
PHP 5.4 is obviously a bridge too far on this one requiring PHPUnit 4 which can't support the PHPUnit 6 changes. Annotations for the exceptions expected may be the better route.

**Comment by padraic** [comment]
Will fail as expected for PHP 5.3.3 to be removed per #28

**Comment by padraic** [comment]
Talk about knocking the rust off something. Three PHPUnit versions, and six PHP versions later..

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
         "padraic/humbug_get_contents": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.0"
+        "phpunit/phpunit": "^4.0|^5.7|^6.0"
```
Comment (theofidry) [inline composer.json]: can't we kill PHPUnit 4.x?

Comment (xabbuh) [inline composer.json]: `^4.8.35` was the first 4.x release that included the FC layer.

Comment (padraic) [inline composer.json]: @theofidry Oh, it's dead after next version here. I'm just loathe to leave downstream without a SHA256 option, even if they are still on EOL PHP versions. Once we're done there, we move directly to whatever we determine as min PHP for Humbug 2.0.

**Comment by omissis** [comment]
Ah, I haven't noticed this PR before submitting mine (https://github.com/humbug/phar-updater/pull/32). I've dropped 4.x, bumped 5.x to 5.6(as it's the oldest that seem to be supporting the FC layer) and added a few more options in the build matrix.

One note about this PR: the expectedException annotation is no longer considered a best practice(https://thephp.cc/news/2016/02/questioning-phpunit-best-practices).

**Comment by omissis** [comment]
@padraic @theofidry I guess this one can be closed now that #32 has been merged.

**Comment by padraic** [comment]
Yep, just checking it merges to no changes ;)

**Pull request closed without merging**